### PR TITLE
Update for jax >= 0.1.55

### PR DESCRIPTION
**Pull request by fehiepsi**
The new version of jax renames `_promote_args_like` to `_promote_args_inexact`, so we need to rename and pump the requirement for master branch.

### CODE REVIEW

**Comment by fehiepsi** [comment]
Blocked by https://github.com/google/jax/issues/1875

File: test/test_distributions_util.py [hunk 1]
```diff
@@ -17,70 +15,8 @@
...
-
-@pytest.mark.parametrize('x, y, grad1, grad2', [
-    (np.array([1., 1., 1.]), np.array([1., 2., 3.]),
-     np.log(np.array([1, 2, 3])), np.array([1., 0.5, 1./3])),
-    (np.array([1.]), np.array([1., 2., 3.]),
-     np.sum(np.log(np.array([1, 2, 3]))), np.array([1., 0.5, 1./3])),
-    (np.array([1., 2., 3.]), np.array([2.]),
-     np.log(np.array([2., 2., 2.])), np.array([3.])),
-    (np.array([0.]), np.array([0., 0.]),
-     np.array([-float('inf')]), np.array([0., 0.])),
```
Comment (fehiepsi) [inline test/test_distributions_util.py]: Using upstream `xlogy` only makes the test for `grad2` here fail (the upstream version returns `nan`, which is caused by `0 / 0`). I think it should be fine to relax that semantic.

**Comment by fehiepsi** [comment]
@neerajprad I think this is unblocked now. Let's see if tests pass.

**Comment by neerajprad** [comment]
There's just one assertion failure for delta distribution where the resulting tensor isn't a `DeviceArray`.

**Comment by fehiepsi** [comment]
Yes, in new version of JAX, broadcast_to a float will be a float.

**Pull request merged**